### PR TITLE
Bug fix/app api tests

### PR DESCRIPTION
**Pull request by mariereed**
Completed AppApi.test.js , passes all tests. Had to add coverage in AppApi.js for updateAppMetadata and updateAppTypeMetadata.

### CODE REVIEW

**Comment by mariereed** [comment]
The only tests I am concerned about are the testUpdateAppMetadata() and testUpdateAppTypeMetadata(). They don't actually test different combinations of the update inputs, it's not testing the has() part ... I modeled it after testUpdateEntitySetMetaData() from EntityDataModelApi.test.js. Considered acceptable, or shall I write a better test?

File: src/api/AppApi.test.js [hunk 1]
```diff
@@ -176,3 +197,394 @@ function testInstallApp() {
...
+
+    testApiShouldReturnPromise(fnToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectPostRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testUpdateAppMetadata() {
```
Comment (UnsungHero97) [inline src/api/AppApi.test.js:392]: This is intense :) `AppApi` is not the only place where this happens, so generally speaking, we need a better way of testing parameters like this. Unfortunately, I haven't put in the time to figure it out.

I recommend simplifying this down to a single case (not 7), like the other functions you've added. We can come back to this later when we have a few spare cycles to think.

File: src/api/AppApi.test.js [hunk 2]
```diff
@@ -176,3 +197,394 @@ function testInstallApp() {
...
+
+    testApiShouldReturnPromise(fnToTest, validParams7);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams7, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams7, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams7, invalidParams);
+    testApiShouldSendCorrectPostRequest(fnToTest, validParams7, axiosParams7);
+  });
+}
+
+function testUpdateAppTypeMetadata() {
```
Comment (UnsungHero97) [inline src/api/AppApi.test.js]: Same comment as above.

File: src/api/AppApi.test.js [hunk 3]
```diff
@@ -176,3 +197,394 @@ function testInstallApp() {
...
+
+    const fnToTest = AppApi.updateAppConfigPermissions;
+    const permissions = [PermissionTypes.READ, PermissionTypes.WRITE];
+
+    const validParams = [MOCK_ORGANIZATION_DM.id, MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id, permissions];
+    const invalidParams = [
+      INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS, [INVALID_PARAMS_SS, INVALID_PARAMS_SS]];
+    const axiosParams = [
+      `/${UPDATE_PATH}/${MOCK_ORGANIZATION_DM.id}/${MOCK_APP_DM.id}/${MOCK_APP_TYPE_DM.id}`,
+      permissions];
```
Comment (UnsungHero97) [inline src/api/AppApi.test.js]: Closing bracket goes on the next line.

File: src/api/AppApi.test.js [hunk 4]
```diff
@@ -176,3 +197,394 @@ function testInstallApp() {
...
+function testUpdateAppConfigPermissions() {
+
+  describe('updateAppConfigPermissions()', () => {
+
+    const fnToTest = AppApi.updateAppConfigPermissions;
+    const permissions = [PermissionTypes.READ, PermissionTypes.WRITE];
+
+    const validParams = [MOCK_ORGANIZATION_DM.id, MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id, permissions];
+    const invalidParams = [
+      INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS, [INVALID_PARAMS_SS, INVALID_PARAMS_SS]];
```
Comment (UnsungHero97) [inline src/api/AppApi.test.js]: After 3 params, if they don't fit on one line, each param should be on its own line. Closing bracket should be on its own line.

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

**Pull request merged**